### PR TITLE
RSDK-3441 Lock usage of Start and Stop for signaling answerer

### PR DESCRIPTION
**Pull request by benjirewis**
[RSDK-3441](https://viam.atlassian.net/browse/RSDK-3441)

Adds a mutex to keep `Start` and `Stop` exclusive for the signaling answerer. Adds a test to ensure `Start` and `Stop` do not race.

[TestWebStreamImmediateClose](https://github.com/viamrobotics/rdk/blob/main/robot/web/web_test.go#L789) is an rdk test I wrote that starts and immediately stops the web service to ensure there are no races/deadlocks. It occasionally (once every ~50 times) causes a [data race](https://github.com/viamrobotics/rdk/actions/runs/5180043576/jobs/9333712461) in goutils. The [signaling answerer’s](https://github.com/viamrobotics/rdk/blob/main/robot/web/web_test.go#L789) `Stop` method had a race with the signaling answerer’s `Start` method. The former tried to `Wait` for background workers and the latter tried to `Add` a new one with no locking.

[RSDK-3441]: https://viam.atlassian.net/browse/RSDK-3441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: testutils/env.go [hunk 1]
```diff
@@ -30,7 +31,7 @@ func SkipUnlessInternet(tb testing.TB) {
 	tb.Helper()
 	if internetConnected == nil {
 		var connected bool
-		conn, err := net.Dial("tcp", "mozilla.org:80")
+		conn, err := net.DialTimeout("tcp", "mozilla.org:80", 5*time.Second)
```
Comment (benjirewis) [inline testutils/env.go:34]: [driveby] this should probably use `DialTimeout` if you're trying to check internet connectivity. Tests will hang in the absence of internet connection otherwise.

File: rpc/wrtc_signaling_answerer.go [hunk 2]
```diff
@@ -24,6 +24,8 @@ import (
 // directly connected to a Server that will handle the actual calls/connections over WebRTC
 // data channels.
 type webrtcSignalingAnswerer struct {
+	mu sync.Mutex // mu guards the Start and Stop methods so they do not happen concurrently.
```
Comment (edaniels) [inline rpc/wrtc_signaling_answerer.go:27]:
```suggestion
	startStopMu sync.Mutex // mu guards the Start and Stop methods so they do not happen concurrently.
```

**Review by edaniels — APPROVED** [review]
LGTM
